syncer/job.go

```diff
@@ -73,41 +75,48 @@ type job struct {
 	currentLocation binlog.Location // end location of the sql in binlog, for user to skip sql manually by changing checkpoint
 	ddls            []string
 	originSQL       string // show origin sql when error, only DDL now
+
+	eventHeader *replication.EventHeader
 }
 
 func (j *job) String() string {
 	// only output some important information, maybe useful in execution.
 	return fmt.Sprintf("tp: %s, sql: %s, args: %v, key: %s, ddls: %s, last_location: %s, start_location: %s, current_location: %s", j.tp, j.sql, j.args, j.key, j.ddls, j.location, j.startLocation, j.currentLocation)
 }
 
-func newJob(tp opType, sourceSchema, sourceTable, targetSchema, targetTable, sql string, args []interface{}, key string, location, startLocation, cmdLocation binlog.Location) *job {
+func newDMLJob(tp opType, sourceSchema, sourceTable, targetSchema, targetTable, sql string, args []interface{},
+	key string, location, startLocation, cmdLocation binlog.Location, eventHeader *replication.EventHeader) *job {
 	return &job{
-		tp:              tp,
-		sourceTbl:       map[string][]string{sourceSchema: {sourceTable}},
-		targetSchema:    targetSchema,
-		targetTable:     targetTable,
-		sql:             sql,
-		args:            args,
-		key:             key,
-		startLocation:   startLocation,
+		tp:           tp,
+		sourceTbl:    map[string][]string{sourceSchema: {sourceTable}},
+		targetSchema: targetSchema,
+		targetTable:  targetTable,
+		sql:          sql,
+		args:         args,
+		key:          key,
+		retry:        true,
+
 		location:        location,
+		startLocation:   startLocation,
 		currentLocation: cmdLocation,
-		retry:           true,
+		eventHeader:     eventHeader,
```

Comment:
It sames that we only use the `TimeStamp` of the `eventHeader`?

Reply:
need use binlog header's fields in next pr (to update some other metrics